### ISSUE TITLE
Am I crazy or is the Documentation for PrimaryKey AutoIncrement wrong?

### ISSUE THREAD

**Issue by pha3z** [praeclarum/sqlite-net#799]
The wiki example says this:

`public class Stock		
{		
    [PrimaryKey, AutoIncrement]		
    public int Id { get; set; }		
    public string Symbol { get; set; }		
}`

From what I can tell, there's no way that code will even work.  It should be this:

`public class Stock		
{		
    [PrimaryKey, AutoIncrement]		
    public int? Id { get; set; }		
    public string Symbol { get; set; }		
}`

If the integer is not nullable, then sqlite-net has no way to know that you didn't intentionally set Id to a default value of 0.  To get AutoIncrement to work, it has to be null.

Am I wrong on this?  If not, I'm going to fix the wiki.

**Comment by pha3z** [comment]
One more thing, wouldn't it also be better to use "long" than "int"?   Under-the-hood, the sqlite table is using a 64-bit value if its following the ROWID pattern described in sqlite docs.  Is this what's going on also with sqlite.net ?

**Comment by pha3z** [comment]
Ok apparently Insert and InsertOrReplace() have different functionality.  And I guess this has been discussed on other posts.  This is very weird.  InsertOrReplace() requires Id to be nullable but Insert does not.  Very confusing.

**Comment by praeclarum** [comment]
Yeah, it's a little weird. You're welcome to use long (or ulong). I just don't bother in the example.

You can think of it as `Insert` always inserts and therefore gets a new PK while `InsertOrReplace` depends on you knowing how things work. Obviously that's not ideal.

I'll keep this open and let others chime in once again. I would like to support non-nullable integer PKs with InsertOrReplace so could special case the value 0? Use that as a placeholder indicating it should always insert?

As far as ROWID is concerned, this library doesn't bubble it up at the ORM level. You're welcome to use it in queries. I did always mean to add an attribute so you can map a property to it...
